### PR TITLE
chore: remove deprecated items

### PR DESCRIPTION
**Pull request by Yun-Kim**
This PR adds a note to upgrade to 2.x in `upgrading.rst` and removes all deprecated items slated for removal in 2.0.0. This includes:

- `DD_GEVENT_PATCH_ALL`: no special configuration is now necessary to make `ddtrace-run` work with gevent.
- `DD_AWS_TAG_ALL_PARAMS`: the boto/botocore/aiobotocore integrations no longer collect all API parameters by default.
- `DD_REMOTECONFIG_POLL_SECONDS`: replaced by `DD_REMOTE_CONFIG_POLL_INTERVAL_SECONDS`
- ASM deprecated constants including ``APPSEC_ENABLED``, ``APPSEC_JSON``, ``APPSEC_EVENT_RULE_VERSION``, ``APPSEC_EVENT_RULE_ERRORS``,
    ``APPSEC_EVENT_RULE_LOADED``, ``APPSEC_EVENT_RULE_ERROR_COUNT``, ``APPSEC_WAF_DURATION``, ``APPSEC_WAF_DURATION_EXT``,
    ``APPSEC_WAF_TIMEOUTS``, ``APPSEC_WAF_VERSION``, ``APPSEC_ORIGIN_VALUE``, ``APPSEC_BLOCKED``,
    ``IAST_JSON``, ``IAST_ENABLED``, ``IAST_CONTEXT_KEY``. These constants were meant for private use only and should not affect existing code.
- ``ddtrace.contrib.grpc.constants.GRPC_PORT_KEY``: replaced by `ddtrace.ext.net.TARGET_PORT`
- ``ddtrace.ext.cassandra.ROW_COUNT``, ``ddtrace.ext.mongo.ROW_COUNT``, ``ddtrace.ext.sql.ROW_COUNT``: replaced by `ddtrace.ext.db.ROWCOUNT`
- `ddtrace.filters.TraceCiVisibilityFilter`: removed as this was for private use only and does not affect existing code.
- `ddtrace.contrib.starlette.get_resource` and `ddtrace.contrib.starlette.span_modifier` and `ddtrace.contrib.fastapi.span_modifier`: the fastapi and starlette integrations now provide the full route and not just mounted route for sub-applications by default.
- `ddtrace.contrib.starlette.config['aggregate_resources']` and `ddtrace.contrib.fastapi.config['aggregate_resources']`: the starlette and fastapi integrations no longer have the option to aggregate resources as this occurs by default now.
- `DD_TRACE_OBFUSCATION_QUERY_STRING_PATTERN`: replaced by `DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP`.

Additionally, the `pep562` dependency and references to it have been removed as it is no longer needed after dropping support for Python < 3.7.

Note that `DD_CALL_BASIC_CONFIG` and `DD_LOG_FORMAT` are removed in #6612 which includes a subtle change in functionality.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by emmettbutler** [comment]
@Yun-Kim how would you like me to approach this review? Since the base isn't 1.x I'm not entirely sure what standards to apply. Maybe I should wait until the base is 1.x to do a review?

**Comment by Yun-Kim** [comment]
@emmettbutler I've cleaned up the PR description to be more clear on what's being changed, and even though I technically don't need an approval to merge into the base branch `2.x-dev` (our current working branch before we cut `2.x`), I'd still love a review!

Ideally #6612 gets merged into this first before we merge this into `2.x-dev`, but I've separated this from #6612 for easier review.

File: releasenotes/notes/remove-deprecated-items-2.0-89eac06472c59554.yaml
Comment (emmettbutler) [inline releasenotes/notes/remove-deprecated-items-2.0-89eac06472c59554.yaml:2]: This could probably use another pass once the change is finalized to make sure it includes everything

File: tests/snapshots/tests.contrib.starlette.test_starlette.test_schematization[service_schema4]_rest.json
Comment (emmettbutler) [inline tests/snapshots/tests.contrib.starlette.test_starlette.test_schematization[service_schema4]_rest.json:5]: I wouldn't expect the removal of deprecated stuff to cause any external behavior (like the emitted spans captured in snapshots) to change. How's this related?

Comment (Yun-Kim) [reply tests/snapshots/tests.contrib.starlette.test_starlette.test_schematization[service_schema4]_rest.json:5]: This is because the test in question `test_schematization` explicitly set `aggregate_resources=False`, but that is no longer an allowed behavior (we will always aggregate resources by default), meaning the asserted resource name should include the `{}` encapsulating the route name. This PR should include some more context: https://github.com/DataDog/dd-trace-py/pull/3612

File: ddtrace/contrib/boto/__init__.py
Comment (majorgreys) [inline ddtrace/contrib/boto/__init__.py:27]: ?

**Review by majorgreys — COMMENTED** [review]
quick check

File: ddtrace/contrib/boto/__init__.py
Comment (Yun-Kim) [reply ddtrace/contrib/boto/__init__.py:27]: This is a weird github diff, line 27 (previously line 43) is still remaining.

File: ddtrace/contrib/grpc/constants.py
Comment (brettlangdon) [inline ddtrace/contrib/grpc/constants.py:31]: can we get rid of this `__getattr__`?

File: ddtrace/constants.py
Comment (brettlangdon) [inline ddtrace/constants.py:1]: we can get rid of this too, right?

File: ddtrace/contrib/grpc/constants.py
Comment (brettlangdon) [reply ddtrace/contrib/grpc/constants.py:31]: and `ensure_pep562`

File: ddtrace/ext/cassandra.py
Comment (brettlangdon) [inline ddtrace/ext/cassandra.py:12]: can we get rid of `__getattr__` and `ensure_pep562`?

File: ddtrace/ext/mongo.py
Comment (brettlangdon) [inline ddtrace/ext/mongo.py:11]: samsies

File: ddtrace/ext/sql.py
Comment (brettlangdon) [inline ddtrace/ext/sql.py:81]: here too?

**Review by brettlangdon — COMMENTED** [review]
minor nits about `__getattr__`/`ensure_pep562`, otherwise lgtm

I did not review the release notes yet.

File: ddtrace/ext/mongo.py
Comment (mabdinur) [inline ddtrace/ext/mongo.py:4]:
```suggestion
```

File: pyproject.toml
Comment (mabdinur) [inline pyproject.toml:47]: Not sure if this in the scope of this PR but we can simplify the dependencies required (drop all references to py<=3.7).

```suggestion
    "protobuf>=3",
```

Comment (Yun-Kim) [reply pyproject.toml:47]: Yup, this is done in #6626 here: https://github.com/DataDog/dd-trace-py/pull/6626/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711

**Review by mabdinur — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

File: tests/internal/remoteconfig/test_remoteconfig.py
Comment (mabdinur) [inline tests/internal/remoteconfig/test_remoteconfig.py:26]: Is this from rebase? Removing this import feels a bit odd. It should either cause syntax or linting errors.

File: ddtrace/internal/remoteconfig/utils.py
Comment (mabdinur) [inline ddtrace/internal/remoteconfig/utils.py:1]: Same here. This seems like a merge conflict that should be resolved in the 2.x branch.

**Review by mabdinur — APPROVED** [review]

File: tests/internal/remoteconfig/test_remoteconfig.py
Comment (Yun-Kim) [reply tests/internal/remoteconfig/test_remoteconfig.py:26]: This was a fix for a poorly done rebase 😓

File: ddtrace/internal/remoteconfig/utils.py
Comment (Yun-Kim) [reply ddtrace/internal/remoteconfig/utils.py:1]: This was also a fix for a poorly done rebase 😓

**Pull request merged**
